pkgs/games/multimc/run.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
```

Comment:
This kind of impurity is disgusting, worse than a fixed-output derivation. I don't think we should be allowing something like this in nixpkgs. If you want to control the version of the software you are distributing to users, please provide an upstream flake for others to use instead.

Reply:
I think disgusting is a bit harsh here. You'll find other self-updating software in nixpkgs already, like Steam. I don't think it's reasonable to expect all software to conform perfectly, and one of NixOS's advantages is that it can still find a place for that software to fit in.

Ultimately whether to include this or not should be a question of practicality, and not some ideal of purity. The question is whether we inconvenience people by making them add an unofficial overlay, or give them the convenience of an official package in nixpkgs. I would prefer the latter.

Reply:
Steam is self-updating in nixpkgs because valve only distributes a few latest versions of the client iirc. Steam breaks every once in a while as it starts depending on new libraries as a result of that. Discord is a good example of self-updating software we have to "work" around but can still provide pinned revisions for because upstream provides a history of old versions.

Reply:
> Steam is self-updating in nixpkgs because valve only distributes a few latest versions of the client iirc.

And it is often the case that the last few versions of MultiMC are the only ones that actually still work. I don't see how that situation is any different just because the previous broken versions are still distributed. Who would want to use software that is no longer functional?

As for Discord, having to wait for packagers to update it whenever they decide to "break" the previous version is a major pain point for me and lots of other people and I would appreciate if that self updated too.

I expect my NixOS system to continue functioning if I change nothing, and these packages that end up with broken old versions do not allow that.

Reply:
> > Steam is self-updating in nixpkgs because valve only distributes a few latest versions of the client iirc.
>
> And it is often the case that the last few versions of MultiMC are the only ones that actually still work. I don't see how that situation is any different just because the previous broken versions are still distributed.

The situation is different because Steam is not FOSS. MultiMC goes against the ideas of FOSS software by imposing restrictions like this on package repositories.

Restrictions on branding material is a little annoying, but understandable.

Trying to add an unreproducible FOSS package to a package repository that focuses on being reproducible, however? That sure is something else.

Reply:
> The situation is different because Steam is not FOSS. MultiMC goes against the ideas of FOSS software by imposing restrictions like this on package repositories.
> 
> Restrictions on branding material is a little annoying, but understandable.

All of the packaging restrictions imposed *are* branding restrictions. If you remove the branding, you can package the software however you like, as PolyMC has done. Maybe this goes against your ideas of FOSS software, but it's in line with the license, which does not grant anyone the rights to use trademarks, only the code. In fact, Firefox, which I think most people would agree is free software, has similar requirements regarding branding.

> Trying to add an unreproducible FOSS package to a package repository that focuses on being reproducible, however? That sure is something else.

To me, one of the great benefits of nixpkgs is that a lot of software that is stuck in the AUR on Arch, or in some overlay on Gentoo, is right here in the official repository. That's why I've made this PR instead of just making a 3rd party flake. If this is not welcome here, then I will just do it the less convenient way with a flake.

Reply:
I don't see any branding issues here, only that you're not actually packaging multimc in a reproducible way. The download of the binary should be locked to a specific revision. I imagine discussion around this topic would go in the same direction regardless of the linux distribution's repos you'd be trying to introduce such a package into.

Reply:
> The download of the binary should be locked to a specific revision.

I agree. In particular there is a huge difference between e.g. firefox-bin and steam-bin which both don't modify the binary itself. Sure steam downloads and runs games if you instruct it to but it doesn't just run arbitrary code out of nowhere.

This in turn makes this pretty useless as then you could also build from source.